LanguageFeatures/Patterns/matching_record_A02_t01.dart

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion At runtime, a pattern is matched against a value. This determines
+/// whether or not the match fails and the pattern refutes the value. If the
+/// match succeeds, the pattern may also destructure data from the object or
+/// bind variables.
+///
+/// Refutable patterns usually occur in a context where match refutation causes
+/// execution to skip over the body of code where any variables bound by the
+/// pattern are in scope. If a pattern match failure occurs in an irrefutable
+/// context, a runtime error is thrown.
+///
+/// To match a pattern p against a value v:
+/// ...
+/// Record:
+/// i. If the runtime type of v is not a subtype of the required type of p, then
+///   the match fails.
+/// ii. For each field f in p, in source order:
+///   a. Access the corresponding field in record v as r.
+///   b. Match the subpattern of f against r. If the match fails, the record
+///     match fails.
+/// iii. The match succeeds if all field subpatterns match.
+///
+/// @description Checks that for each field `f` in `p` in a source order the
+/// corresponding field in record `v` is matched against subpattern
```

Comment:
Sounds like this one is the correct test for doing all those pattern matching cases where we test values in the subpatterns!